sympy/core/relational.py

```diff
@@ -302,9 +302,19 @@ def equals(self, other, failing_expression=False):
 
     def _eval_simplify(self, **kwargs):
         from .add import Add
+        from sympy.core.expr import Expr
         r = self
         r = r.func(*[i.simplify(**kwargs) for i in r.args])
         if r.is_Relational:
+            if r.lhs.is_Set and r.rhs.is_Set:
+                lhs = r.lhs.is_subset(r.rhs)
+                rhs = r.rhs.is_subset(r.lhs)
+                result = fuzzy_and([lhs, rhs])
+                if result is None:
+                    return self
+                return result
```

Comment:
We can shortcut one of the `is_subset` calls if we do
```
result = fuzzy_and(a.is_subset(b) for a, b in [(lhs, rhs), (rhs, lhs)])
```
When this gives true or false it will return Python's `True` or `False` but it should return sympy's `S.true` or `S.false` because `Equality` is a symbolic `Boolean` so any simplification should also give a Boolean. Note that `S.true/false` are instances of `Basic` and have the associated methods like `subs` etc.